src/components/routing/routers/GameRouter.js

```diff
@@ -17,36 +18,51 @@ const GameRouter = props => {
 
       {/* menu */}
       <Route exact path={`${props.base}/menu`}>
-        <GameMenu/>
+          <GameGuard>
+              <GameMenu/>
+          </GameGuard>
```

Comment:
did you include the game guard here again in order to trigger the check for the token? if so, simple and clean fix! could have probably also used useEffect on history or push url events, but this works too

Reply:
Yes, exactly. I mean, GameGuard does only one thing anyway, so no need to study useEffect on history or push events ;)